demos/interactive_face_detection_demo/main.cpp

```diff
@@ -59,340 +47,524 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+        } else {
+            std::cout<< "To close the application, press 'CTRL+C' here" << std::endl; 
+        }
+
+        cv::VideoWriter videoWriter;
+
+        Timer timer;
 
-                face->headPoseEnable((headPoseDetector.enabled() &&
-                                      i < headPoseDetector.maxBatch));
-                if (face->isHeadPoseEnabled()) {
-                    face->updateHeadPose(headPoseDetector[i]);
+        stream.start();
+
+        while (stream.running())
+        {
+            timer.start("total");
+
+            if (!stream.pull(std::move(out_vector))) {
+                std::cout<<"End of streaming" << std::endl;
+                if(FLAGS_loop_video) {
+                    stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(FLAGS_i));
+                    stream.start();
+                } else if (!FLAGS_no_wait) {
+                    std::cout << "No more frames to process!" << std::endl;
+                    cv::waitKey(0);
                 }
+            } else {
+                if (!FLAGS_no_show && emotions_enable && !FLAGS_no_show_emotion_bar) {
+                    visualizer->enableEmotionBar(frame.size(), {"neutral",
+                                                                "happy",
+                                                                "sad",
+                                                                "surprise",
+                                                                "anger"});
+                }
+
+                //  Postprocessing
+                std::list<Face::Ptr> prev_faces;
 
-                face->landmarksEnable((facialLandmarksDetector.enabled() &&
-                                       i < facialLandmarksDetector.maxBatch));
-                if (face->isLandmarksEnabled()) {
-                    face->updateLandmarks(facialLandmarksDetector[i]);
+                if (!FLAGS_no_smooth) {
+                    prev_faces.insert(prev_faces.begin(), faces.begin(), faces.end());
                 }
 
-                faces.push_back(face);
-            }
+                faces.clear();
 
-            //  Visualizing results
-            if (!FLAGS_no_show || !FLAGS_o.empty()) {
-                out.str("");
-                out << "Total image throughput: " << std::fixed << std::setprecision(2)
-                    << 1000.f / (timer["total"].getSmoothedDuration()) << " fps";
-                cv::putText(prev_frame, out.str(), cv::Point2f(10, 45), cv::FONT_HERSHEY_TRIPLEX, 1.2,
-                            cv::Scalar(255, 0, 0), 2);
+                // For every detected face
+                for (size_t i = 0; i < face_hub.size(); i++) {
+                    Face::Ptr face;
 
-                // drawing faces
-                visualizer->draw(prev_frame, faces);
+                    cv::Rect rect = face_hub[i] & cv::Rect({0, 0}, frame.size());
+                    faceProcessing(frame, face, rect,
```

Comment:
what "processing" means here?

Looking at this code, it is not clear what happens inside.

Reply:
Suggest to rename the functions to make the code more clear.

Reply:
`void faceProcessing()` -> `void faceDataUpdate()`
Other cases are similar